src/gluonts/model/r_forecast/_predictor.py

```diff
@@ -178,38 +229,70 @@ def predict(
         self,
         dataset: Dataset,
         num_samples: int = 100,
+        intervals: Optional[List] = None,
         save_info: bool = False,
         **kwargs,
-    ) -> Iterator[SampleForecast]:
-        for entry in dataset:
-            if isinstance(entry, dict):
-                data = entry
-            else:
-                data = entry.data
```

Comment:
I’m still unsure how this was supposed to work…

Reply:
:)

Yea, it is always dict and `data` = `entry`.